File: backend/app/db/models/emotion_log.py
from sqlalchemy import Column, Integer, String, Text, BigInteger, ForeignKey, Float
from sqlalchemy.types import JSON
from sqlalchemy.orm import relationship
from app.db.base_class import Base

class EmotionLog(Base):
    __tablename__ = "emotion_logs"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"))
    timestamp = Column(BigInteger, nullable=False, index=True)
    detected_mood = Column(String, nullable=True)
    source_text = Column(Text)
    source_feature = Column(String)
    sentiment_score = Column(Float, nullable=True)
    key_emotions_detected = Column(JSON, nullable=True)

    user = relationship("User")

File: app/src/main/java/com/psy/deardiary/data/model/EmotionLog.kt
package com.psy.deardiary.data.model

data class EmotionLog(
    val id: Int,
    val timestamp: Long,
    val detectedMood: String?,
    val sourceText: String,
    val sourceFeature: String,
    val sentimentScore: Float?,
    val keyEmotionsDetected: List<String>?
)
